src/cluster_legacy.c

```diff
@@ -5378,18 +5399,33 @@ void clusterCron(void) {
     /* Clear so clusterNodeCronHandleReconnect can count the number of nodes in PFAIL. */
     server.cluster->stats_pfail_nodes = 0;
     /* Run through some of the operations we want to do on each cluster node. */
+    /* We want to start from the random position within cluster nodes just to be safe. */
+    int total = dictSize(server.cluster->nodes);
+    int skip = rand() % (total == 0 ? 1 : total);
+
+    /* First iterator: from skip point to end */
     di = dictGetSafeIterator(server.cluster->nodes);
+    int i = 0;
+
+    /* Skip first 'skip' entries */
+    while (i++ < skip && (de = dictNext(di)) != NULL) {
+    }
```

Comment:
I would prefer us taking this functionality to the dictionary level to avoid these random pointer access. It would be nice if we can introduce an API like `dictGetRandomIterator(d)` and avoid maintaining any information on the caller side.

Reply:
Yeah, I think having a random iterator as a generic construct is very useful.